BuildMasterAutomation/Functions/New-BMEnvironment.ps1

```diff
@@ -0,0 +1,49 @@
+
+function New-BMEnvironment
+{
+    <#
+    .SYNOPSIS
+    Creates a new environment in a BuildMaster instance.
+
+    .DESCRIPTION
+    The `New-BMEnvironment` creates a new environment in BuildMaster. Pass the name of the environment to the `Name` parameter. Names may only contain letters, numbers, spaces, periods, underscores, or dashes.
+    
+    Every environment must have a unique name. If you create a environment with a duplicate name, you'll get an error.
+
+    Environments can't be deleted. Deleted environments are just disabled/inactive. If you need to reactivate/enable a disabled environment, use `Enable-BMEnvironment`. If you try to create a new environment with the same name as an inactive environment, you'll get an error.
+    
+    Pass a session object representing the instance of BuildMaster to use to the `Session` parameter. Use the `New-BMSession` function to create session objects.
+
+    This function uses BuildMaster's infrastructure management API.
+
+    .EXAMPLE
+    New-BMEnvironment -Session $session -Name 'DevNew'
+
+    Demonstrates how to create a new environment.
+    #>
+    [CmdletBinding(SupportsShouldProcess)]
+    param(
+        [Parameter(Mandatory)]
+        # An object representing the instance of BuildMaster to connect to. Use `New-BMSession` to create session objects.
+        [object]$Session,
+
+        [Parameter(Mandatory)]
+        [ValidatePattern('^[A-Za-z][A-Za-z0-9_-]*(?<![_-])$')]

```

Comment:
The docs in the `Description` say `Names may only contain letters, numbers, spaces, periods, underscores, or dashes.` but this regex does not allow for spaces or periods.